src/test/components/CallTreeSidebar.test.js

```diff
@@ -179,12 +188,24 @@ describe('CallTreeSidebar', function () {
       queryByText,
       getAllByText,
       funcNamesDict: { A, B, C },
+      getState,
+      rerenderContainer,
     } = setup(getProfileWithSubCategories());
     selectNode([A, B, C]);
     expect(queryByText('FakeSubCategoryC')).not.toBeInTheDocument();
+    expect(container.firstChild).toMatchSnapshot();
+    expect(getState().urlState.profileSpecific.sidebarOpenCategories).toEqual(
+      new Set()
+    );
 
     const layoutCategory = getAllByText('Layout')[0];
+
     fireFullClick(layoutCategory);
+    rerenderContainer();
```

Comment:
I tried to debug why this is still needed and frankly I don't have an answer. From my testing, it looks like React doesn't commit the virtual dom changes to the actual DOM, but I've never seen that happening before. It may be because the Breakdown component is using fragments a lot, including a fragment in a loop, and maybe (but not sure) this has an impact here.
Also this works fine in a real browser...

My random though is that maybe we should have retrieved the open categories in the CallTreeSidebar component and passed it to the Breakdown as a prop. But let's not do that now, maybe we can do that later as a refactoring. Also it would be more correct because we could pass different values to different breakdowns.

Reply:
I have the same opinion.